grommet/grommet
# Add selected theme property for DataTable (#7343)

#### What does this PR do?

It introduces support for recognising an additional theme property to customise the background colour of selected records in the DataTable component.

#### Where should the reviewer start?

The proposed schema extension to DataTable related theme properties, then onward to the DataTable source.

#### What testing has been done on this PR?

I've added three new snapshot tests for DataTable covering the cases where I saw selected record styling being pertinent.

#### How should this be manually tested?

Ensure that the new theme property is set and that a DataTable is configured to support selection.

#### Do Jest tests follow these best practices?


#### Any background context you want to provide?

None.

#### What are the relevant issues?

Closes #7343.

#### Screenshots (if appropriate)

I've modified some DataTable stories to illustrate the visual scope of the new theme property.

Single row selection:

![image](https://github.com/user-attachments/assets/ed7c7008-d5d2-413f-9bc9-ec7cedc69bfd)

Grouped row selection:

![image](https://github.com/user-attachments/assets/023a0709-4cba-4288-b833-64415c6e309a)

Row detail parent selection:

![image](https://github.com/user-attachments/assets/0081d416-163d-4eef-8ff0-024025b381db)

#### Do the grommet docs need to be updated?

Yes, should indicate the avaiability of this new theme property.

#### Should this PR be mentioned in the release notes?

Probably, for maintainers to decide.

#### Is this change backwards compatible or is it a breaking change?

The new theme property isn't set by default, so it shouldn't be a breaking change.
